packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -219,18 +219,19 @@ export class Tracing implements InstrumentationListener, SnapshotterDelegate, Ha
         entries.push({ name: path.join('resources', sha1), value: path.join(this._resourcesDir, sha1) });
 
       const sourceEntries: NameValue[] = [];
+      const targetForSources = (params.mode === 'compressTraceAndSources') ? entries : sourceEntries;
       for (const value of state.sources)
-        sourceEntries.push({ name: 'resources/src@' + calculateSha1(value) + '.txt', value });
+        targetForSources.push({ name: 'resources/src@' + calculateSha1(value) + '.txt', value });
```

Comment:
Let's put an if statement here - I find it hard to understand that we push into one of the two arrays and that affects the semantics.

Reply:
Done.